src/main/java/com/randomEventAnalytics/TimeTracking.java

```diff
@@ -100,6 +101,26 @@ public int getNextRandomEventEstimation()
 		return SPAWN_INTERVAL_SECONDS - secondsMod;
 	}
 
+	public boolean isPossibleTimeForRandomEvent()
+	{
+		if (!hasLoggedInLongEnoughForSpawn())
```

Comment:
What are your thoughts about checking if there is a previously logged random here as well, as part of a solve for:
> [...] the shortest is invalid data, because it's the first one. It essentially gives the time since logging instead of the time since the previous random since there was no previous random.

eg
```java
private boolean isEligibleForRandom() { // TODO: Better method name, and maybe closure/lambda in isPossibleTimeForRandomEvent
  int loginTime = getSecondsSinceLogin();
  int secondsMod = loginTime % SPAWN_INTERVAL_SECONDS;
  // If we're within 15 seconds of an event window, it's considered a possible time.
  return secondsMod <= SPAWN_INTERVAL_TIMEFRAME_SECONDS || secondsMod >= SPAWN_INTERVAL_SECONDS - SPAWN_INTERVAL_TIMEFRAME_SECONDS;
}

public boolean isPossibleTimeForRandomEvent() {
  if (this.lastRandomSpawnInstant == null) { // This should determine if there's a previous random.
    return this.isEligibleForRandom();
  }
  // ...
  // your existing checks here
  return this.isEligibleForRandom();
}
```

Though I may be getting too edge-casey; there are probably other existing facets of this class that do not properly account for first time events. 😓